File: Tests/load_balancer/load_balancer_server/load_balancer_executors/CustomHeuristic.cpp
#include "CustomHeuristic.h"

static uint64_t getId();

CustomHeuristic::CustomHeuristic(std::string_view ip, std::string_view port, bool useHTTPS) :
	LoadBalancerHeuristic(ip, port, useHTTPS),
	id(getId())
{

}

uint64_t CustomHeuristic::operator ()() const
{
	return id;
}

uint64_t getId()
{
	static std::atomic_uint64_t id = 0;

	return id++;
}

DECLARE_HEURISTIC(CustomHeuristic)

File: Tests/executors/CSharp/LoadBalancerExecutors/CustomHeuristic.cs
using Framework;

public class CustomHeuristic(string ip, string port, bool useHTTPS) : LoadBalancerHeuristic(ip, port, useHTTPS)
{
	private readonly ulong id = GetId();

	public override ulong Invoke()
	{
		return id;
	}

	private static ulong currentId = 0;

	private static ulong GetId()
	{
		return Interlocked.Add(ref currentId, 1);
	}
}
